tests/test_termui.py

```diff
@@ -485,3 +487,196 @@ def cmd(arg1):
+    ({"prompt": True, "default": True, "flag_value": False}, [], "[Y/n]", "", True),
+    ({"prompt": True, "default": True, "flag_value": False}, [], "[Y/n]", "y", True),
+    ({"prompt": True, "default": True, "flag_value": False}, [], "[Y/n]", "n", False),
+    # default=False
+    ({"prompt": True, "default": False, "flag_value": True}, [], "[y/N]", "", False),
+    ({"prompt": True, "default": False, "flag_value": True}, [], "[y/N]", "y", True),
+    ({"prompt": True, "default": False, "flag_value": True}, [], "[y/N]", "n", False),
+    ({"prompt": True, "default": False, "flag_value": False}, [], "[y/N]", "", False),
+    ({"prompt": True, "default": False, "flag_value": False}, [], "[y/N]", "y", True),
+    ({"prompt": True, "default": False, "flag_value": False}, [], "[y/N]", "n", False),
+    # default=None
+    # XXX
+    # (
+    #     {"prompt": True, "default": None, "flag_value": True},
+    #     [],
+    #     "[y/n]",
+    #     "",
+    #     False,
+    # ),
+    ({"prompt": True, "default": None, "flag_value": True}, [], "[y/n]", "y", True),
+    ({"prompt": True, "default": None, "flag_value": True}, [], "[y/n]", "n", False),
+    # XXX
+    # (
+    #     {"prompt": True, "default": None, "flag_value": False},
+    #     [],
+    #     "[y/n]",
+    #     "",
+    #     False,
+    # ),
+    ({"prompt": True, "default": None, "flag_value": False}, [], "[y/n]", "y", True),
+    ({"prompt": True, "default": None, "flag_value": False}, [], "[y/n]", "n", False),
+    # If the flag_value is None, the flag behave like a string flag, whatever the
+    # default is.
+    ({"prompt": True, "default": True, "flag_value": None}, [], "[True]", "", "True"),
+    ({"prompt": True, "default": True, "flag_value": None}, [], "[True]", "y", "y"),
+    ({"prompt": True, "default": True, "flag_value": None}, [], "[True]", "n", "n"),
+    (
+        {"prompt": True, "default": False, "flag_value": None},
+        [],
+        "[False]",
+        "",
+        "False",
+    ),
+    ({"prompt": True, "default": False, "flag_value": None}, [], "[False]", "y", "y"),
+    ({"prompt": True, "default": False, "flag_value": None}, [], "[False]", "n", "n"),
+    # XXX ({"prompt": True, "default": None, "flag_value": None}, [], "", "", "False"),
+    ({"prompt": True, "default": None, "flag_value": None}, [], "", "y", "y"),
+    ({"prompt": True, "default": None, "flag_value": None}, [], "", "n", "n"),
+]
+
+
+@pytest.mark.parametrize(
+    ("opt_decls", "opt_params", "args", "prompt", "input", "expected"),
+    # Boolean flag prompt cases.
+    [("--flag/--no-flag", *case_params) for case_params in BOOLEAN_FLAG_PROMPT_CASES]
+    # Non-boolean flag prompt cases.
+    + [("--flag", *case_params) for case_params in FLAG_VALUE_PROMPT_CASES],
+)
+def test_flag_value_prompt(
+    runner, opt_decls, opt_params, args, prompt, input, expected
+):
+    """Covers concerns raised in issue https://github.com/pallets/click/issues/1992."""
```

Comment:
Describe what the test does in the docstring, the link should be extra context only.